### ISSUE TITLE
Custom Analyzer does not work

### ISSUE THREAD

**Issue by abtpst** [elastic/elasticsearch#13280]
I am trying to create a new index in Elastic Search as follows

  PUT /abtest2
    {
      "settings":{

       "analysis":{

         "char_filter":{

           "replaceJunk":{

             "type":"pattern_replace",
             "pattern":"[^\\d+]",
             "replacement":""
           }
         },

          "analyzer":{

     "dateAnalyzer":{

     "type":"custom",
     "tokenizer":"standard",
     "char_filter":["replaceJunk"]
     }
    }

   }
  },

      "mappings":{

            "fix":{

                "properties": {

                   "Date": {
                      "type": "date",
                      "analyzer": "dateAnalyzer"
                   }
                }
             }
        }
    }

so basically, i want to create a custom analyzer for my Date field which might have some non numeric characters coming in. Now, i try to pass a value like

PUT /abtest2/fix/5
{
     "Date":"14186196000-0005"
}
but this gives me an error

    {
   "error": "MapperParsingException[failed to parse [Date]]; nested: MapperParsingException[failed to parse date field [14186196000-0005], tried both date format [dateOptionalTime], and timestamp number with locale []]; nested: IllegalArgumentException[Invalid format: \"14186196000-0005\" is malformed at \"00-0005\"]; ",
   "status": 400
}

shouldn't this field have been analyzed and transformed ?

please advise

**Comment by clintongormley** [comment]
Hi @abtpst 

Analyzers are for strings, not for date fields.  In 2.0 this mapping would give you an exception.

**Comment by abtpst** [comment]
thanks Clinton, but even if i change it to string, it still does not remove the - character

**Comment by abtpst** [comment]
even if i set the pattern to be just -

"replaceJunk":{ "type":"pattern_replace", "pattern":"-", "replacement":"" }

**Comment by clintongormley** [comment]
@abtpst Please ask questions like these in the forum http://discuss.elastic.co/
